lib/location/method_gnss.c

```diff
@@ -717,6 +717,20 @@ static void method_gnss_positioning_work_fn(struct k_work *work)
 int method_gnss_location_get(const struct location_method_config *config)
 {
 	int err;
+	enum lte_lc_system_mode mode;
+
+	err = lte_lc_system_mode_get(&mode, NULL);
+	if (err) {
+		LOG_ERR("Could not read LTE system mode. Error: %d", err);
+		return err;
+	}
+
+	if (mode != LTE_LC_SYSTEM_MODE_GPS &&
```

Comment:
Looks good to me. One could also check that the system mode is not LTE_LC_SYSTEM_MODE_GPS is A-GPS or P-GPS is enabled. But I don't know if that is necessary. In such case I believe A-GPS request would just fail and GNSS position would be returned anyway. So definitely the check you just implemented is more important because there is no way to get a fix without GNSS being enabled.

Reply:
I have just added such checks, and moved the checks to their own static inline functions. Let me know if these changes are acceptable!